### PR TITLE
fix(@ngtools/webpack): fix aot builds using npm packages that have la…

### PR DESCRIPTION
**Pull request by maleetz**
Currently, code splitting derived from routes with the loadChildren property only works
if the resolved source file is a *.ts file in the source directory of your angular-cli
project. If your project uses an npm package that has a route with loadChildren, the
build will resolve the lazy loaded module as a *.d.ts definition file and then attempt
to find an *.d.ts.ngfactory.ts file in the generated AOT directory, which does not exist.
This fixes the path generation logic so the build will look for a *.ngfactory.ts file
in $$_gendir/node_modules.

### CODE REVIEW

File: packages/@ngtools/webpack/src/plugin.ts [hunk 1]
```diff
@@ -461,8 +461,8 @@ export class AotPlugin implements Tapable {
             if (this.skipCodeGeneration) {
               this._lazyRoutes[k] = lazyRoute;
             } else {
-              const lr = path.relative(this.basePath, lazyRoute.replace(/\.ts$/, '.ngfactory.ts'));
-              this._lazyRoutes[k + '.ngfactory'] = path.join(this.genDir, lr);
+              const lr = path.relative(this.basePath, lazyRoute.replace(/(\.d)?\.ts$/, '.ngfactory.ts'));
+              this._lazyRoutes[k + '.ngfactory'] = path.join(this.genDir, path.join('/', lr));
```
Comment (hansl) [inline packages/@ngtools/webpack/src/plugin.ts]: I understand the change on the line before (which LGTM), but I'm curious for why do you join `/` here and what does that mean for Windows users?

Comment (maleetz) [inline packages/@ngtools/webpack/src/plugin.ts]: Thanks for the comment! I'm thinking I should have used path.sep instead to be platform agnostic. Will update the PR shortly.

Comment (maleetz) [inline packages/@ngtools/webpack/src/plugin.ts]: And the purpose of the join is to ensure lr cannot traverse up the directory tree and escape the $$_gendir directory. Assuming the following:

- basePath: '/project/src'
- genDir: '/project/src/$$_gendir
- lazyRoute: '/project/node_modules/package/lazy.module.ngfactory.ts'.

lr would then be '../node_modules/package/lazy.module.ngfactory.ts' and then the resulting _lazyRoutes path would end up being '/project/src/node_modules/package/lazy.module.ngfactory.ts', but the actual AOT factory file would be in '/project/src/$$_gendir/node_modules/package/lazy.module.ngfactory.ts'

**Comment by filipesilva** [comment]
@hansl can you re-review?

**Comment by maleetz** [comment]
Fixed the linter issue but I couldn't reproduce the issue from e2e test case tests/build/rebuild.

Running "tests/build/rebuild" (9 of 9)...
  ==========================================================================================
  Running `ng "serve"` [silent, matching(/webpack: bundle is now VALID|webpack: Compiled successfully./)]...
  CWD: /private/tmp/angular-cli-e2e-117321-19025-dm4gnr.nwwtmaemi/test-project
  ==========================================================================================
  Running `touch "src/main.ts"`...
  CWD: /private/tmp/angular-cli-e2e-117321-19025-dm4gnr.nwwtmaemi/test-project
  ==========================================================================================
  Running `ng "generate" "module" "lazy" "--routing"`...
  CWD: /private/tmp/angular-cli-e2e-117321-19025-dm4gnr.nwwtmaemi/test-project
    installing module
      create src/app/lazy/lazy-routing.module.ts
      create src/app/lazy/lazy.module.ts
----
Last step took 27.14s...

**Comment by maleetz** [comment]
Any suggestions on what I need to do to avoid this test failure? It seems it is examining commits that it probably shouldn't.

Examining 1331 commit(s) between HEAD and master
INVALID COMMIT MSG: "Initial commit"
 => ERROR: The commit message does not match the format of "<type>(<scope>): <subject> OR revert: type(<scope>): <subject>"

**Pull request closed without merging**